### PR TITLE
fix anonymous boolean issue

### PR DESCRIPTION
**Pull request by mwickett**
Instead of trying to parse the string right at the field level, I just cast it on form submit.

### CODE REVIEW

File: components/form/index.tsx [hunk 1]
```diff
@@ -59,9 +59,15 @@ export default function StoryForm() {
       onSubmit={async (values: LoginFormInputs, actions) => {
         try {
           actions.setSubmitting(true)
+          const anonymousValue = values.anonymous
+          const payload = {
+            ...values,
+            // cast the string anonymous value to boolean
+            anonymous: JSON.parse(anonymousValue),
+          }
```
Comment (mwickett) [inline components/form/index.tsx:67]: Less complex solve - I leave the value as a string until submission when it gets cast to a boolean. `yup.boolean` must be parsing 'true' as valid? 🤷

File: pages/api/stories/index.ts [hunk 2]
```diff
@@ -64,10 +64,6 @@ async function handlePost(req: NextApiRequest, res: NextApiResponse): Promise<vo
     // Strip empty values
     if (req.body[k] === '') {
       return acc
-    }
-    // Parse anonymous to boolean
-    else if (k === 'anonymous') {
-      acc[k] = JSON.parse(req.body[k])
```
Comment (mwickett) [inline pages/api/stories/index.ts:70]: 🔪

File: components/form/index.tsx (hunk 1 above)
Comment (davesteinberg) [reply components/form/index.tsx:67]: Geez yup, get it together.

File: pages/api/stories/index.ts (hunk 2 above)
Comment (davesteinberg) [reply pages/api/stories/index.ts:70]: This is in `lib/api/stories.ts` now, and it's more complicated because Typescript. Byeee!

**Review by davesteinberg — APPROVED** [review]

**Comment by mwickett** [comment]
Ok, committed a change and I'll test on the preview here first before I merge.

**Pull request merged**